### ISSUE TITLE
How to set default 'add filter' match type

### ISSUE THREAD

**Issue by dgleba** [railsadminteam/rails_admin#3001]
I have searched and read a lot but I can't find any hints on how to do this.

When I 'add filter' per the picture below the default match is ... which does nothing.  See picture below.
 
I want to set the default matching to contains. I want contains to be showing and be used with no user interaction. If they want to change it from that they can. It will be the default though rather than ...

Can someone point to docs or tell me how to do that?

![image](https://user-images.githubusercontent.com/4622715/37174885-dcb992e4-22e5-11e8-9d94-39c943765586.png)

**Comment by ChrissyFlemingVTS** [comment]
We noticed a similar issue. From my internal team:

It used to be, when you added a search parameter in rails admin, it would default the parameter type to "contains."

Starting on the 26th of February (best I can tell), it left the parameter type blank, forcing the user to select one prior to searching:

This is a minor change but it is an extra step for an action that we do upwards of 50 times a day.

Can this be changed back so that it defaults to "contains"?

**Comment by mattruzicka** [comment]
Our team would also like to be able to specify a default match type, and not always "contains". So having the default configurable would be great. We may be able to look into adding this behavior, but no promises.

**Comment by coorasse** [comment]
I solved it by defining the following in a `config/initializers/rails_admin/main_helper.rb` file:

```
require 'rails_admin/main_helper'

module RailsAdmin
  module MainHelper
    alias_method :original_ordered_filter_options, :ordered_filter_options

    def ordered_filter_options
      @ordered_filter_options ||= original_ordered_filter_options.tap do |ordered_filter_options|
        ordered_filter_options.each { |options| options[:operator] ||= :like if options[:type] == :string }
      end
    end
  end
end
```

I basically override the original method and set the operator to `:like` if is not set already.

**Comment by AlexBelov** [comment]
I was able to solve this by adding

```
$(document).on("click", "#filters a", function (e) {
  e.preventDefault();
  $.filters.append({
    label: $(this).data("field-label"),
    name: $(this).data("field-name"),
    type: $(this).data("field-type"),
    value: $(this).data("field-value"),
    operator: $(this).data("field-type") == 'string' ? 'like' : $(this).data("field-operator"),
    select_options: $(this).data("field-options"),
    required: $(this).data("field-required"),
    index: $.now().toString().slice(6, 11),
    datetimepicker_options: $(this).data("field-datetimepicker-options"),
  });
});
```

to the `app/assets/javascripts/rails_admin/custom/filter-box.js`

**Comment by codealchemy** [comment]
Following https://github.com/railsadminteam/rails_admin/pull/3318 ([released in `2.1.0`](https://github.com/railsadminteam/rails_admin/blob/master/CHANGELOG.md#210---2021-02-28)) you can set the `default_filter_operator` for a given field in your Rails Admin config.
